### PR TITLE
Feature/govt inspection pallets restrict CRUD for shipped pallets

### PR DESCRIPTION
**Pull request by ViljoenFritz**
Summary of this change:
Restrict crud on govt_inspection_pallets for shipped pallets

CHANGELOG entry:

### Added
- [Updated test_helper to take user args](https://github.com/NoSoft-SA/nspack/pull/392/commits/ebd402065c81e56d3b8ceb57f5ac9bc76e356a3f)
- [Added Tests](https://github.com/NoSoft-SA/nspack/pull/392/commits/6c42f313dfb8ea90c35aa72a1ea5c7649520fe06)
### Changed
- [Refactored pass_govt_inspection_pallet](https://github.com/NoSoft-SA/nspack/pull/392/commits/19f82109e1055fc8445c30444c0a3ed3dcbad163)
- [Updated Task Checkers for pallets and govt_inspection_pallets](https://github.com/NoSoft-SA/nspack/pull/392/commits/0148775c4d189f8624e646c9205c633ed79a5d73)
### Fixed
- [fix govt_inspection_pallets grid for pallets without sequences](https://github.com/NoSoft-SA/nspack/pull/392/commits/aa82ad45e1d3a241285555d7fff1dac8c711c596)

### CODE REVIEW

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb [hunk 1]
```diff
@@ -3,12 +3,14 @@
 module FinishedGoodsApp
   module TaskPermissionCheck
     class GovtInspectionPallet < BaseService
-      attr_reader :task, :entity
-      def initialize(task, govt_inspection_pallet_id = nil)
+      attr_reader :task, :entity, :args
+      def initialize(task, *ids, **args)
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:7]: With the :create check you dont send a id, but I still want to check the pallet_id for shipped

File: lib/finished_goods/repositories/govt_inspection_repo.rb [hunk 2]
```diff
@@ -129,6 +129,8 @@ def govt_inspection_sheet_packhouse_code(id)
     # --------------------------------------------------------------------------
 
     def find_govt_inspection_pallet(id)
+      return nil unless id.is_a? Integer
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/finished_goods/repositories/govt_inspection_repo.rb:132]: Had a bug last night, I probably over reacted

File: lib/finished_goods/interactors/govt_inspection_pallet_interactor.rb [hunk 3]
```diff
@@ -126,5 +127,10 @@ def govt_inspection_pallet(id)
     def govt_inspection_pallet_grid_row(id)
       dataminer_grid_row(:govt_inspection_pallets, id)
     end
+
+    def check!(check, *id, **args)
```
[the remark this answers is not in the document]
Comment (ViljoenFritz) [reply lib/finished_goods/interactors/govt_inspection_pallet_interactor.rb:131]: I want id to be optional

[the remark this answers is not in the document]
Comment (jcsjcs) [reply lib/finished_goods/interactors/govt_inspection_pallet_interactor.rb:131]:
```ruby
[124] pry(main)> def check!(check, id = nil, args: {})
[124] pry(main)*   p id, args
[124] pry(main)* end  
=> :check!
[125] pry(main)> check!('Look, no id', args: { a: 123 })
nil
{:a=>123}
=> [nil, {:a=>123}]
[126] pry(main)> check!('Look, an id!', 23, args: { a: 123 })
23
{:a=>123}
=> [23, {:a=>123}]
[127] pry(main)> check!('Look, nothing at all')
nil
{}
=> [nil, {}]
```
Just do
```ruby
check!(:create, args: res.to_h)
```

File: lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (jcsjcs) [reply lib/finished_goods/task_permission_checks/govt_inspection_pallet.rb:7]: But you always call it with an id (which can be nil) from `check!`.

**Pull request merged**